Kstests daily scenario v2

rvykydal opened the pull request:
This PR is addressing issues from review of https://github.com/rhinstaller/permian/pull/10

Related kickstart-tests PR integrating permian: https://github.com/rhinstaller/kickstart-tests/pull/691

velezd commented on libpermian/plugins/kickstart_test/__init__.py:
```diff
@@ -151,15 +161,94 @@ def __init__(self, testRuns, crcList, arch):
             raise UnsupportedConfiguration('architecture', self.arch)
 
+        if self.event.kstestParams:
+            try:
```
platform must be in KstestParamsStructure, so this is not necessary

rvykydal replied:
Updated

velezd commented on libpermian/plugins/kickstart_test/__init__.py:
```diff
@@ -151,15 +161,94 @@ def __init__(self, testRuns, crcList, arch):
+                self.platform = self.event.kstestParams.platform
+            except KeyError:
+                pass
+            try:
```
urls must be in KstestParamsStructure, so this is not necessary

rvykydal replied:
Updated

velezd reviewed (commented):
I've noticed that on line 262 you should raise an exception, the testrun will try to execute the workflow even without boot iso

rvykydal commented:
> I've noticed that on line 262 you should raise an exception, the testrun will try to execute the workflow even without boot iso

Good catch, actually I was hoping you will not notice and I will update it later - because the launcher script actually has a fallback if it doesn't find iso on expected location it downloads a rawhide iso - but I think the correct behavior is to require the iso location to be configured. I'll update it.
@velezd: I didn't find any suitable existing exception in libpermian/exceptions.py so I guess I'll create a new local one, say "MissingBootIso" ?

velezd commented:
@rvykydal sounds good and I already implemented silent exceptions so you could add the exception to it. Also, it's ok if you fix this in next pull request.

rvykydal commented:
> @rvykydal sounds good and I already implemented silent exceptions so you could add the exception to it. Also, it's ok if you fix this in next pull request.

Addressed in the last commit of the PR https://github.com/rhinstaller/permian/pull/14/commits/a221fe9cca1d5b7af37489fd72a1ad86ce8e3da5

rvykydal commented on libpermian/plugins/kickstart_test/__init__.py:
```diff
@@ -151,15 +173,88 @@ def __init__(self, testRuns, crcList, arch):
             LOGGER.info(f"Architecture {self.arch} is not supported.")
             raise UnsupportedConfiguration('architecture', self.arch)
 
+        if self.event.kstestParams:
```
Agreed

rvykydal replied:
Upated in https://github.com/rhinstaller/permian/pull/14/commits/ec25169b51a8550e1391996d3af9ca77e6987cb6

pholica commented on libpermian/plugins/kickstart_test/__init__.py:
```diff
@@ -237,10 +332,24 @@ def execute(self):
+        if self.platform:
+            command = command + ['--platform', self.platform]
+
+        if self.defaults_file_path:
```
Please add a comment here that this is in fact "URLs overrides" or something like that.

rvykydal replied:
https://github.com/rhinstaller/permian/pull/14/commits/a1ec2bac503f1817cf400c65440dbac9205b6808

rvykydal commented on libpermian/plugins/kickstart_test/__init__.py:
```diff
@@ -237,10 +332,24 @@ def execute(self):
-        command = self.runner_command + tests
+        command = self.runner_command
+
+        if self.platform:
```
Updated timeout setting (in the original commit)

pholica replied:
The `self.platform` is taken from `self.event.kstestParams.platform` which is now mandatory field in 'kstestParams' event structure. I believe the `if` statement here is not necessary unless there's a reason to handle `False` values such as empty string, I guess this was forgotten/lost in the discussion.

rvykydal commented:
@pholica ah, I forgot the redundant "if platform:", now I updated the last commit.
